### PR TITLE
Use the Zipkin URLConnectionReporter from new zipkin-reporter libs…

### PR DESCRIPTION
**Pull request by nicmunroe**
… rather than copy-pasting from brave

### CODE REVIEW

**Comment by nicmunroe** [comment]
@adriancole - everything looks a lot nicer after getting rid of the copy-paste code and using the zipkin-reporter lib. This should make it trivial for users to take advantage of new reporters as they show up. Thanks for pulling the reporter stuff out into a separate lib!

File: build.gradle [hunk 1]
```diff
@@ -64,7 +64,8 @@ javadocJar.enabled = false
 ext {
     slf4jVersion = '1.7.7'
     servletApiVersion = '3.1.0'
-    zipkinVersion = '1.7.0'
+    zipkinCoreVersion = '1.7.0'
+    zipkinReporterVersion = '0.2.0'
```
Comment (adriancole) [inline build.gradle]: ps this api will thrash a while (pre 1.0), but it is unlikely people will have dependents on it in the same classpath.
choice is to accept the hazard or shade/shadow it until 1.0

File: wingtips-zipkin/README.md [hunk 2]
```diff
@@ -26,9 +26,22 @@ Parameters for this basic `WingtipsToZipkinLifecycleListener` constructor:
...
 * **`postZipkinSpansBaseUrl`** - The base URL of the Zipkin server. This should include the scheme, host, and port (if non-standard for the scheme). e.g. `http://localhost:9411`, or `https://zipkinserver.doesnotexist.com/`.
 
-## Using Native Zipkin Brave `SpanCollector`s
+## Using Native Zipkin `Reporter`s
+
+The default `ZipkinSpanSender` implementation provided by this submodule (used under the hood when you follow the Quickstart instructions above) for sending spans to the Zipkin server uses `URLConnectionReporter` from the official Zipkin `zipkin-reporter-urlconnection` library. This should work decently for many use cases, however it is limited to HTTP transport. You can trivially use any of the other reporters provided by the [zipkin-reporter-java](https://github.com/openzipkin/zipkin-reporter-java) repository by pulling in the dependencies necessary to have access to the `Reporter` implementation you want to use, and then initialize with the following:
```
Comment (adriancole) [inline wingtips-zipkin/README.md]: ahh well if documenting this, than shade/shadow won't work :) I might suggest holding off on merging this until monday, as I'm refactoring some of the key problems in the current design. Most important is kafka has a blocking api problem, which means the "Sender" design I'm working on will include a health check. At bootstrap, folks can invoke that to make sure kafka is available as opposed to having potentially a minute of blocking then an exception!

File: wingtips-zipkin/src/main/java/com/nike/wingtips/zipkin/util/ZipkinSpanSenderDefaultHttpImpl.java [hunk 3]
```diff
@@ -1,203 +0,0 @@
...
-     *     however it may lack some of the flexibility your project requires (e.g. around connection pooling). You can extend this class and override
-     *     this method to use a different HTTP client for sending spans to Zipkin if desired.
-     * </p>
-     * <p>
-     *     This code was derived from the Zipkin/Brave repository's brave-spancollector-http module v3.9.1
-     *     (https://github.com/openzipkin/brave/blob/master/brave-spancollector-http/src/main/java/com/github/kristofa/brave/http/HttpSpanCollector.java)
-     *     and licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0).
-     * </p>
-     */
-    protected void sendSpans(byte[] spanListJsonPayloadBytes) throws IOException {
```
Comment (adriancole) [inline wingtips-zipkin/src/main/java/com/nike/wingtips/zipkin/util/ZipkinSpanSenderDefaultHttpImpl.java]: new Sender api I'm defining will have this signature

```
public interface Sender extends Component {

  /**
   * Sends a list of encoded spans to a transport such as http or Kafka.
   *
   * <p>Usually, this involves enqueueing them for transport over http or Kafka. The typical end
   * recipient is a zipkin {@link Collector}.
   *
   * <p>Note: Eventhough there's a callback, there's no guarantee implementations won't block.
   * Accordingly, this method should not be called on the operation being measured's thread.
   *
   * @param spans encoded list of spans. Usually encoded with {@link zipkin.Codec#writeSpans(List)}
   * @param callback signals either completion or failure
   */
  void sendSpans(byte[] spans, Callback callback);
}
```

File: build.gradle (hunk 1 above)
Comment (nicmunroe) [inline build.gradle]: Understood. Not too concerned for the moment. If it becomes problematic I can take the shade/shadow steps you mentioned.

File: wingtips-zipkin/README.md (hunk 2 above)
Comment (nicmunroe) [inline wingtips-zipkin/README.md]: Sounds good. Will hold off on merging for the moment.

File: build.gradle (hunk 1 above)
Comment (nicmunroe) [inline build.gradle]: I changed my mind - I think I'll decline this PR for now and give you time to get the reporter libs into a more stable state. The old copy-paste code works fine for now, and I'll keep this commit around. Should be easy to update to whatever you end up with when the time comes.

**Comment by nicmunroe** [comment]
Closing this PR without merging for now. After the zipkin-reporter libs stabilize a little I'll update and reopen.

**Pull request closed without merging**